web-search-engines/langchain4j-web-search-engine-searchapi/src/main/java/dev/langchain4j/web/search/searchapi/SearchApiClient.java

```diff
@@ -0,0 +1,87 @@
+package dev.langchain4j.web.search.searchapi;
+
+import static com.google.gson.FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES;
+
+import java.io.IOException;
+import java.time.Duration;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import lombok.Builder;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class SearchApiClient {    
+    private final SearchApiAPI searchapiAPI;
+    private final String engine;
+    private final String apiKey;
+        
+    /* Indicates whether to log requests and their responses. Defaults to false. */
+    private final boolean logRequests;
+    
+    private static final Gson GSON = new GsonBuilder()
+            .setFieldNamingPolicy(LOWER_CASE_WITH_UNDERSCORES)
+            .setPrettyPrinting()
+            .create();
+
+    @Builder
+    public SearchApiClient(final String baseUrl,
+    					final String apiKey,
+    					final String engine,
+    					final Duration timeout, 
+    					final boolean logRequests) {
+        this.apiKey = apiKey;
+        this.engine = engine;
+        this.logRequests = logRequests;
+        
+        final OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder()
+                .callTimeout(timeout)
+                .connectTimeout(timeout)
+                .readTimeout(timeout)
+                .writeTimeout(timeout);
+        
+        if (this.logRequests) {
+            final HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
+            interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+            okHttpClientBuilder.addInterceptor(interceptor);
+        }
+
+        final Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(baseUrl)
+                .client(okHttpClientBuilder.build())
+                .addConverterFactory(GsonConverterFactory.create(GSON))
+                .build();
+
+        this.searchapiAPI = retrofit.create(SearchApiAPI.class);
+    }
+
+    public SearchApiResponse search(final SearchApiRequest searchRequest) {
```

Comment:
```suggestion
    SearchApiResponse search(final SearchApiRequest searchRequest) {
```
since `SearchApiRequest` has default visibility

Reply:
Fixed.